src/Server/CertificateReloader.cpp

```diff
@@ -93,32 +109,57 @@ void CertificateReloader::tryLoad(const Poco::Util::AbstractConfiguration & conf
     }
     else
     {
-        bool cert_file_changed = cert_file.changeIfModified(std::move(new_cert_path), log);
-        bool key_file_changed = key_file.changeIfModified(std::move(new_key_path), log);
-        std::string pass_phrase = config.getString("openSSL.server.privateKeyPassphraseHandler.options.password", "");
+        auto it = data.end();
+        auto i = data_index.find(prefix);
+        if (i != data_index.end())
+            it = i->second;
+        else
+        {
+            data.push_back(MultiData(ctx));
+            --it;
+            data_index[prefix] = it;
+        }
+
+        bool cert_file_changed = it->cert_file.changeIfModified(std::move(new_cert_path), log);
+        bool key_file_changed = it->key_file.changeIfModified(std::move(new_key_path), log);
+        std::string pass_phrase = config.getString(prefix + "privateKeyPassphraseHandler.options.password", "");
 
         if (cert_file_changed || key_file_changed)
         {
-            LOG_DEBUG(log, "Reloading certificate ({}) and key ({}).", cert_file.path, key_file.path);
-            data.set(std::make_unique<const Data>(cert_file.path, key_file.path, pass_phrase));
-            LOG_INFO(log, "Reloaded certificate ({}) and key ({}).", cert_file.path, key_file.path);
+            LOG_DEBUG(log, "Reloading certificate ({}) and key ({}).", it->cert_file.path, it->key_file.path);
+            it->data.set(std::make_unique<const Data>(it->cert_file.path, it->key_file.path, pass_phrase));
+            LOG_INFO(log, "Reloaded certificate ({}) and key ({}).", it->cert_file.path, it->key_file.path);
         }
 
         /// If callback is not set yet
         try
         {
-            if (init_was_not_made)
-                init();
+            if (!ctx)
+            {
+                ctx = Poco::Net::SSLManager::instance().defaultServerContext()->sslContext();
+                it->ctx = ctx;
+            }
+
+            if (it->init_was_not_made)
+                init(&*it);
         }
         catch (...)
```

Comment:
what kind of exceptions are we mute here? Should we continue to work when such errors happened?

Reply:
Honestly - no idea, try{}catch(...) already was around init method.